<?xml version="1.0"?>
<launch>

    <arg name="agent1" default="a1_950"/>
    <arg name="agent2" default="hsr"/>

    <arg name="agent1_pose_topic" default="$(arg agent1)/global_pose"/>
    <arg name="agent2_pose_topic" default="$(arg agent2)/global_pose"/>
    <arg name="agent1_map_topic" default="$(arg agent1)/fov_map"/>
    <arg name="agent2_map_topic" default="$(arg agent2)/fov_map"/>
    <arg name="num_agents" default="2"/>

    <param name="AGENT1_POSE_TOPIC" value="$(arg agent1_pose_topic)"/>
    <param name="AGENT2_POSE_TOPIC" value="$(arg agent2_pose_topic)"/>
    <param name="AGENT1_MAP_TOPIC" value="$(arg agent1_map_topic)"/>
    <param name="AGENT2_MAP_TOPIC" value="$(arg agent2_map_topic)"/>
    <param name="NUM_AGENT" value="$(arg num_agents)"/>

  <node name="map_server" pkg="map_server" type="map_server" args="$(find tmp)/maps/map.yaml"/>

    <node name="scaled_map_converter" pkg="search_service" type="scaled_map_converter" output="screen">
        <!--<remap from="/map" to="/walrus/map"/>-->
    </node>
    <node name="multisearch_action" pkg="search_service" type="mtsp_server_forward" output="screen">
        <remap from="/agent_0_path" to="/a1_950/search_agent_manager/waypoint_plan"/>
        <remap from="/agent_1_path" to="/hsr/search_agent_manager/waypoint_plan"/>
    </node>
    <node name="getunknowns" pkg="visual_perception" type="prediction_server" output="screen"/>
    <!--<node name="tsp_solve_action" pkg="search_service" type="tsp_server" output="screen"/>-->
    <node name="polygon_client" pkg="search_service" type="plugin_client" output="screen"/>

    <node name="agent_0_tspsolve" pkg="search_service" type="stsp_server" output="screen"/>
    <node name="agent_1_tspsolve" pkg="search_service" type="stsp_server" output="screen"/>

    <arg name="target_nav_action_0" default="$(arg agent1)/move_base"/>
    <arg name="target_nav_action_1" default="$(arg agent1)/move_base"/>

    <arg name="target_path_0" default="agent_0_path"/>
    <arg name="target_path_1" default="agent_1_path"/>


    <node name="agent_0_getpath" pkg="search_service" type="get_visualpath_server" output="screen"/>
    <node name="agent_1_getpath" pkg="search_service" type="get_visualpath_server" output="screen"/>

    <!-- Run move_base for getting global plans in forwardpath_server -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find tmp)/config/turtlebot/local_costmap_params.yaml" command="load"/>
        <rosparam file="$(find tmp)/config/turtlebot/global_costmap_params.yaml" command="load"/>
        <rosparam file="$(find tmp)/config/turtlebot/move_base_params.yaml" command="load" />
        <rosparam file="$(find tmp)/config/turtlebot/dwa_local_planner_params_burger.yaml" command="load" />
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
        <param name="base_global_planner" value="global_planner/GlobalPlanner"/>
    </node>


</launch>
